WordPress/gutenberg
# Add stylelint for SCSS linting

Supersedes https://github.com/WordPress/gutenberg/pull/1151. Closes #8618.

Adds the `npm run lint-css` and `npm run lint-css:fix` commands for SCSS linting, and configures `npm run lint` to automatically run `npm run lint-css`.

Linting is performed by [stylelint](https://github.com/stylelint/stylelint). I've configured stylelint to extend [stylelint-config-wordpress](https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress) which enforces the [WordPress CSS coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/).

To save time and reduce merge conflicts, I've disabled every rule that isn't totally trivial to fix (e.g. with a find and replace). We can work towards re-enabling these rules one by one over the medium term.

To test: 

1. `npm run lint-css`
2. Open the demo post and check for visual regressions
3. Check the diff and make sure `stylelint --fix` or one of my regular expressions didn't completely backfire
